client/ratings_reviews/components/ReviewFormModal.jsx

```diff
@@ -61,80 +81,120 @@ class ReviewFormModal extends React.Component {
         name: '',
         email: '',
         photos: [],
-        characteristics: {}
+        characteristics: {},
+        photoCount: 0,
+        show: false,
+        showUploadPhotosButton: true
       }, this.props.closeModal);
     }).catch((error) => {
       console.log(error);
     });
   }
 
+  toggleModal() {
+    this.setState({
+      show: !this.state.show
+    });
+  }
+
+  validateForm() {
```

Comment:
maybe better as an external helper function? maybe not...

Reply:
I might be able to put validateForm in helpers. Will look into it!